### PR TITLE
[node] Add overloads for process.emitWarning

### PR DESCRIPTION
**Pull request by Qix-**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/process.html#process_process_emitwarning_warning_options
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by sanderkoenders — APPROVED** [review]
LGTM

**Comment by Qix-** [comment]
_beep boop_ No idea what I should do to fix the `jsdom` failures @typescript-bot. They're unrelated and even happen on local master. Looks like something got merged prematurely, or something changed and now we're seeing the side effect here.

There's nothing to await from me.

**Comment by Qix-** [comment]
@typescript-bot Uhhh please don't. The CI has nothing to do with this PR. Tests on master are broken. This is why I hate bots on PRs...

Bump. Can someone advise on how to either fix CI or merge this? Due to how Typescript prevents you from overloading classes and interfaces, this is blocking us from some work being completed...

**Comment by ThomasdenH** [comment]
This has been a problem for some time now.

**Comment by Archcry** [comment]
You can try to just change someting minor to trigger a new build and hope for the best xD

File: types/node/index.d.ts [hunk 1]
```diff
@@ -783,7 +791,9 @@ declare namespace NodeJS {
         chdir(directory: string): void;
         cwd(): string;
         debugPort: number;
+        emitWarning(warning: string | Error, options?: WarningOptions): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: I think {options} should be mandatory here as the variant with only one arg is already covered by another overload.

File: types/node/index.d.ts [hunk 2]
```diff
@@ -783,7 +791,9 @@ declare namespace NodeJS {
...
         emitWarning(warning: string | Error, name?: string, ctor?: Function): void;
+        emitWarning(warning: string | Error, name?: string, code?: string, ctor?: Function): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: I think `name` and `code` should be mandatory but `string | undefined` here as the single arg version is already covered.

**Comment by Flarna** [comment]
similar as in #31709.
@andy-ms @sandersn Any hints how to fix the jsdom issue? is it caused by dtslint?

**Comment by Qix-** [comment]
@typescript-bot Oh my god come on - I have a full time job and can't get to this until tonight....

**Comment by dcecile** [comment]
@typescript-bot closed mine at #31709 too (same jsdom failure), not sure who has permissions to override the robot overlord 🤖

**Pull request closed without merging**